spec/spec.md

```diff
@@ -343,6 +350,8 @@ The lifetime of the returned `auth_token` is not defined. If a privileged method
 
 Dapp endpoints should make every effort possible to [verify the authenticity](#dapp-identity-verification) of the presented identity. While the `dapp_uri` parameter is optional, it is strongly recommended - without it, the wallet endpoint may not be able to verify the authenticity of the dapp.
 
+The `cluster` parameter allows the dapp endpoint to select a specific Solana cluster with which to interact. This is relevant for both [`sign_transaction`](#sign_transaction), where a wallet may refuse to sign transactions without a currently valid blockhash, and for [`sign_and_send_transaction`](#sign_and_send_transaction), where the wallet endpoint must know which cluster to submit the transactions to. This parameter would normally be used to select a cluster other than `mainnet_beta` for dapp development and testing purposes. Under normal circumstances, this field should be omitted, in which case the wallet endpoint is free to either automatically select an appropriate cluster, or to present the user with a cluster selection option.
```

Comment:
```suggestion
The `cluster` parameter allows the dapp endpoint to select a specific Solana cluster with which to interact. This is relevant for both [`sign_transaction`](#sign_transaction), where a wallet may refuse to sign transactions without a currently valid blockhash, and for [`sign_and_send_transaction`](#sign_and_send_transaction), where the wallet endpoint must know which cluster to submit the transactions to. This parameter would normally be used to select a cluster other than `mainnet-beta` for dapp development and testing purposes. Under normal circumstances, this field should be omitted, in which case the wallet endpoint is free to either automatically select an appropriate cluster, or to present the user with a cluster selection option.
```

Reply:
> Under normal circumstances, this field should be omitted, in which case the wallet endpoint is free to either automatically select an appropriate cluster, or to present the user with a cluster selection option.

This being optional is weird because it reduces the explicitness of the API. Why would the dapp want to leave the selection up to the wallet in any circumstance? It knows where it wants to send transactions, and it needs to confirm them against an RPC node on the same cluster.

It also doesn't get a value back from the wallet to indicate which cluster the wallet chose, leaving it to guess or try to infer from the recent blockhash?

Reply:
Good point. Making the default `mainnet-beta`

Reply:
Done

Reply:
> Under normal circumstances, this field should be omitted, in which case the wallet endpoint is free to either automatically select an appropriate cluster, or to present the user with a cluster selection option.

How does the dapp find out which cluster the wallet chose?

Reply:
This comment is on an old version of the changes. The latest text reads:
> This parameter would normally be used to select a cluster other than `mainnet-beta` for dapp development and testing purposes. Under normal circumstances, this field should be omitted, in which case the wallet endpoint will interact with the `mainnet-beta` cluster.

Reply:
Ahh, my bad. Thanks!